blogs/qserial.md

```diff
@@ -0,0 +1,243 @@
+Hey AngularJS devs!  Let's talk about executing async tasks serially with promises.
+
+First, we notice .then allows our success/fail callbacks to return a promise.  $q treats success/fail callbacks that return promises specially.  We'll look at two examples to see how $q treats success/fail functions that return promises differently from those that don't.
+<!--more-->
+
+Normal success function:
+<pre>
+$q.when()  
+.then(function () { console.log("foo"); })
+.then(function () { console.log("bar"); });
+</pre>
+<pre>
+foo
+bar
+</pre>
+<br/>
+Success function that returns a promise:
+<pre>
+$q.when()
+.then(function () { 
+  console.log("foo"); 
+  var bazDeferred = $q.defer();
```

Comment:
bazDeferred is a "Deferred".  I really dislike the name.  I'd prefer it to just be called task (eventhough it's not really a task).  I think the following reads much more clearly:
```
var bazTask = $q.task();
doStuff()
bazTask.complete();

var bazTask = $q.defer();
doStuff();
bazTask.resolve();
```